app/src/main/java/org/simple/clinic/home/patients/PatientsScreen.kt

```diff
@@ -67,6 +79,11 @@ open class PatientsScreen(context: Context, attrs: AttributeSet) : RelativeLayou
         .subscribe { uiChange -> uiChange(this) }
   }
 
+  override fun onDetachedFromWindow() {
+    super.onDetachedFromWindow()
+    loggedOutAlertDialog = null
+  }
```

Comment:
I have an an aversion to using Fragments in any form. :P

I can change it to use a fragment dialog if it makes sense. It doesn't in this case to me, because the dialog does not encapsulate any business logic for now.

Reply:
Unfortunately, we need something to handle the lifecycle of the dialog. A fragment will do this automatically for us. It also allows us to avoid keeping a reference here.